ipwb/replay.py

```diff
@@ -179,6 +179,126 @@ def showMementosForURIRs_sansJS():
+            key = key.strip("/,")
+
+    return keys
+
+
+def bin_search(iter, key, datetime=None):
+    # Read line encompassing current position
+    ln = iter.readline()
+
+    while ipwbUtils.isCDXJMetadataRecord(ln):
+        ln = iter.readline()
+
+    surtk, rest = ln.split(maxsplit=1)
+    if key == surtk:
+        return [ln]
+
+    # If further searching required...
+    left = 0
+    # Go to end of seek stream
+    iter.seek(0, 2)
+    right = iter.tell()  # Current position
+
+    lines = set()  # Prevents dupes
+    while (right - left > 1):
+        mid = (right + left) // 2
+        iter.seek(mid)
+        iter.readline()  # Purge rest of current line
+        line = iter.readline()  # Read the next full line
+
+        if len(line) == 0:
+            right = mid
+            continue
+
+        surtk, rest = line.split(maxsplit=1)
+        datetimeK = rest.split()[0].decode()
+        if surtk[-1:] == b'/':
+            surtk = surtk[0:-1]
+
+        # TODO: find a more elegant way for comparison than manually
+        # trimming the last two chars off of the surtk
+
+        if key == surtk:
+            if datetime and datetime != datetimeK:
+                # Rm other close matches, exact found
+                lines.clear()
+                lines.add(line)
+
+                break
+            lines.add(line)
+            # Iterate further to get lines after selection point
+            nextLine = iter.readline()
+            while nextLine:
+                surtk, rest = nextLine.split(maxsplit=1)
+                datetimeK = rest.split()[0].decode()
+                if surtk[-1:] == b'/':
+                    surtk = surtk[0:-1]
+
+                if key == surtk:
+                    lines.add(nextLine)
+
+                nextLine = iter.readline()
+
+            # Continue searching until find first instance
+            right = mid
+        elif key > surtk:
+            left = mid
+        else:
+            right = mid
+
+    # Convert uniq set to list then sort and return
+    ret = sorted(list(lines))
+
+    return ret
+
+
+def run_batchlookup(surt, filename, datetime=None):
+    mobj = open(filename, "rb")
+    iter = mobj
+
+    fobj = open(filename, "rb")
+    for line in fobj:
+        res = lookup(iter, surt, datetime)
+        if res:
+            return res
+    fobj.close()
+    mobj.close()
+
+
+def lookup(iter, surt, datetime=None):
+    for idx, key in enumerate(lookup_keys(surt)):
+        res = bin_search(iter, key.encode(), datetime)
+        if res:
+            return res
+
+
+def getCDXJLinesWithURIR(urir, indexPath, datetime=None):
+    """ Get all CDXJ records corresponding to a URI-R """
+    if not indexPath:
+        indexPath = ipwbUtils.getIPWBReplayIndexPath()
+    indexPath = getIndexFileFullPath(indexPath)
+
+    # Convert URI-R to surt
+    surtedURIR = surt.surt(urir, path_strip_trailing_slash_unless_empty=True)
+
+    res = run_batchlookup(surtedURIR, indexPath, datetime)
```

Comment:
After removing the `run_batchlookup` function, we need to think about it and reimplement in a way that is efficient and works.